### PR TITLE
RPN-851

### PR DESCRIPTION
**Pull request by mystdeim**

### CODE REVIEW

File: nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/item/meta/MediaMetaService.kt [hunk 1]
```diff
@@ -125,6 +125,7 @@ class MediaMetaService(
                 val r = readers.next()
                 r.setInput(iis, true)
                 try {
+                    println("${r.getWidth(0)}, ${r.getHeight(0)}")
```
Comment (yakischik) [inline nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/item/meta/MediaMetaService.kt:128]: Maybe logger?

Comment (mystdeim) [reply nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/item/meta/MediaMetaService.kt]: Forgot to remove it. Fixed

File: nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/item/meta/MediaMetaService.kt [hunk 2]
```diff
@@ -109,13 +111,11 @@ class MediaMetaService(
...
-            conn.readTimeout = timeout
-            conn.connectTimeout = timeout
-            conn.setRequestProperty("user-agent", "curl/7.73.0")
-            conn.inputStream.use { get(it) }
-        }.blockingToMono()
+        return client(url).get().uri(url)
+            .retrieve()
+            .bodyToFlux(DataBuffer::class.java)
+            .reduce { a, b -> a.factory().join(listOf(a, b)) }
+            .map { get(it.asInputStream()) }
```
Comment (serejke) [inline nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/item/meta/MediaMetaService.kt:118]: Probably we have to pass `releaseOnClosed = true` to the `asInputStream` function call in order to release resources allocated for the bytebuffer.

File: nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/item/meta/MediaMetaService.kt [hunk 3]
```diff
@@ -135,7 +135,24 @@ class MediaMetaService(
         }
     }
 
+    private fun client(url: String): WebClient {
```
Comment (serejke) [inline nft/core/src/main/kotlin/com/rarible/protocol/nft/core/service/item/meta/MediaMetaService.kt:138]: We create `WebClient` for every opensea URL request. Maybe we can use Kotlin's 
```
val openSeaClient: WebClient by lazy { 
WebClient.builder()
     .clientConnector(Proxy.createConnector(timeout, timeout, proxyUrl, true))
     .build()
```
to cache the client

**Pull request merged**